include/led.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Driver specific representation of a LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable time period (in ms) supported by the device */
+	u32_t min_period;
+	/* Maximum acceptable time period (in ms) supported by the device */
+	u32_t max_period;
+	/* Minimum acceptable brightness (in percent) supported by the device */
+	u32_t min_brightness;
+	/* Maximum acceptable brightness (in percent) supported by the device */
+	u32_t max_brightness;
+};
+
+/**
+ * @typedef led_api_blink()
+ * @brief Callback API for blinking a LED
+ *
+ * @see led_blink() for argument descriptions.
+ */
+typedef int (*led_api_blink)(struct device *dev, u32_t led, u32_t period);
+
+/**
+ * @typedef led_api_blink_oneshot()
+ * @brief Callback API for oneshot blink of a LED
+ *
+ * @see led_blink_oneshot() for argument descriptions.
+ */
+typedef int (*led_api_blink_oneshot)(struct device *dev, u32_t led);
+
+/**
+ * @typedef led_api_set_brightness()
+ * @brief Callback API for setting brightness of a LED
+ *
+ * @see led_set_brightness() for argument descriptions.
+ */
+typedef int (*led_api_set_brightness)(struct device *dev, u32_t led,
+				      u32_t value);
+/**
+ * @typedef led_api_on()
+ * @brief Callback API for turning on an LED
+ *
+ * @see led_on() for argument descriptions.
+ */
+typedef int (*led_api_on)(struct device *dev, u32_t led);
+
+/**
+ * @typedef led_api_off()
+ * @brief Callback API for turning off an LED
+ *
+ * @see led_off() for argument descriptions.
+ */
+typedef int (*led_api_off)(struct device *dev, u32_t led);
+
+/**
+ * @brief LED driver API
+ *
+ * This is the mandatory API any LED driver needs to expose.
+ */
+struct led_driver_api {
+	led_api_blink blink;
+	led_api_blink_oneshot blink_oneshot;
+	led_api_set_brightness set_brightness;
+	led_api_on on;
+	led_api_off off;
+};
+
+/**
+ * @brief Blink an LED
+ *
+ * This routine blinks an LED for the given time period continuously
+ *
+ * @param dev LED device
+ * @param led LED channel/pin
+ * @param period Period (in micro second) to blink an LED
+ * @return 0 on success, negative on error
+ */
+static inline int led_blink(struct device *dev, u32_t led, u32_t period)
+{
+	const struct led_driver_api *api = dev->driver_api;
+
+	return api->blink(dev, led, period);
+}
+
+/**
+ * @brief Oneshot blink of an LED
```

Comment:
If this API is unimplemented, let's just leave it out for now. It can be added to the core API when there is a user.

Reply:
Sure, will leave it for now!